quickstarts/microsoft.network/aad-domainservices/main.bicep

```diff
@@ -0,0 +1,180 @@
+
+@description('Subnet prefix')
+param domainServicesSubnetAddressPrefix string = '10.0.0.0/24'
+
+var domainServicesNSGName = '${domainServicesSubnetName}-nsg'
+
+var PSRemotingSlicePIPAddresses = [
+  '52.180.179.108'
+  '52.180.177.87'
+  '13.75.105.168'
+  '52.175.18.134'
+  '52.138.68.41'
+  '52.138.65.157'
+  '104.41.159.212'
+  '104.45.138.161'
+  '52.169.125.119'
+  '52.169.218.0'
+  '52.187.19.1'
+  '52.187.120.237'
+  '13.78.172.246'
+  '52.161.110.169'
+  '52.174.189.149'
+  '40.68.160.142'
+  '40.83.144.56'
+  '13.64.151.161'
+]
+var RDPIPAddresses = [
+  '207.68.190.32/27'
+  '13.106.78.32/27'
+  '13.106.174.32/27'
+  '13.106.4.96/27'
+]
+
+var PSRemotingSliceTIPAddresses = [
+  '52.180.183.67'
+  '52.180.181.39'
+  '52.175.28.111'
+  '52.175.16.141'
+  '52.138.70.93'
+  '52.138.64.115'
+  '40.80.146.22'
+  '40.121.211.60'
+  '52.138.143.173'
+  '52.169.87.10'
+  '13.76.171.84'
+  '52.187.169.156'
+  '13.78.174.255'
+  '13.78.191.178'
+  '40.68.163.143'
+  '23.100.14.28'
+  '13.64.188.43'
+  '23.99.93.197'
+]
+
+resource nsg 'Microsoft.Network/networkSecurityGroups@2020-11-01' = {
+  name: domainServicesNSGName
+  location: location
+  properties: {
+    securityRules: [
+      {
+        name: 'AllowPSRemotingSliceP'
+        properties: {
+          protocol: 'Tcp'
+          sourcePortRange: '*'
+          destinationPortRange: '5986'
+          sourceAddressPrefixes: PSRemotingSlicePIPAddresses
+          destinationAddressPrefix: '*'
+          access: 'Allow'
+          priority: 301
+          direction: 'Inbound'
+        }
+      }
+      {
+        name: 'AllowRDP'
+        properties: {
+          protocol: 'Tcp'
+          sourcePortRange: '*'
+          destinationPortRange: '3389'
+          sourceAddressPrefixes: RDPIPAddresses
+          destinationAddressPrefix: '*'
+          access: 'Allow'
+          priority: 201
+          direction: 'Inbound'
+        }
+      }
+      {
+        name: 'AllowSyncWithAzureAD'
+        properties: {
+          protocol: 'Tcp'
+          sourcePortRange: '*'
+          destinationPortRange: '443'
+          sourceAddressPrefix: '*'
+          destinationAddressPrefix: '*'
+          access: 'Allow'
+          priority: 101
+          direction: 'Inbound'
+        }
+      }
+      {
+        name: 'AllowPSRemotingSliceT'
+        properties: {
+          protocol: 'Tcp'
+          sourcePortRange: '*'
+          destinationPortRange: '5986'
+          sourceAddressPrefixes: PSRemotingSliceTIPAddresses
+          destinationAddressPrefix: '*'
+          access: 'Allow'
+          priority: 302
+          direction: 'Inbound'
+        }
+      }
+    ]
+  }
+}
+
+resource vnet 'Microsoft.Network/virtualNetworks@2020-11-01' = {
+  name: domainServicesVnetName
+  location: location
+  properties: {
+    addressSpace: {
+      addressPrefixes: [
+        domainServicesVnetAddressPrefix
+      ]
+    }
+  }
+  dependsOn: [
```

Comment:
why is this needed?

Reply:
It's there in the original JSON:
```json
    {
      "apiVersion": "2020-11-01",
      "type": "Microsoft.Network/virtualNetworks",
      "name": "[parameters('domainServicesVnetName')]",
      "location": "[parameters('location')]",
      "dependsOn": ["[variables('domainServicesNSGName')]"],
      "properties": {
        "addressSpace": {
          "addressPrefixes": ["[parameters('domainServicesVnetAddressPrefix')]"]
        }
      },
      "resources": [
        {
          "apiVersion": "2020-11-01",
          "type": "subnets",
          "location": "[parameters('location')]",
          "name": "[parameters('domainServicesSubnetName')]",
          "dependsOn": ["[parameters('domainServicesVnetName')]"],
          "properties": {
            "addressPrefix": "[parameters('domainServicesSubnetAddressPrefix')]",
            "networkSecurityGroup": {
              "id": "[variables('nsgRefId')]"
            }
          }
        }
      ]
    }
```

I don't want to remove those myself cause I don't know anything about the RP structures.  if you want me to try removing it, I can, although one successful deployment doesn't prove it's not needed.  So what do you want me to do?

Reply:
ah ok... the vnet doesn't have the dependency, the subnet does - but since the nsg is referenced by the subnet, bicep will create the dependson for the user... so we should remove it.

There's likely to be many more of these cases as we go through the migration.  Might be worth an informal rule to "question" any explicit dependsOn in bicep code.

Reply:
Will remove, then.  I'm removing them when I can find clear reason.  When not, I'm leaving it in and will depend on you to tell me if it's unnecessary.  Thanks.